gossipd/gossip.c

```diff
@@ -599,6 +599,51 @@ static struct io_plan *peer_start_gossip(struct io_conn *conn, struct peer *peer
 			 peer_pkt_out(conn, peer));
 }
 
+static void handle_get_update(struct peer *peer, const u8 *msg)
+{
+	struct short_channel_id schanid;
+	struct node *us;
+	size_t i;
+	const u8 *update;
+
+	if (!fromwire_gossip_get_update(msg, NULL, &schanid)) {
+		status_trace("peer %s sent bad gossip_get_update %s",
+			     type_to_string(trc, struct pubkey, &peer->id),
+			     tal_hex(trc, msg));
+		return;
+	}
+
+	/* We want update than comes from our end. */
+	us = node_map_get(peer->daemon->rstate->nodes, &peer->daemon->id.pubkey);
+	if (!us) {
+		status_trace("peer %s schanid %s but can't find ourselves",
+			     type_to_string(trc, struct pubkey, &peer->id),
+			     type_to_string(trc, struct short_channel_id,
+					    &schanid));
+		update = NULL;
+		goto reply;
+	}
+
+	for (i = 0; i < tal_count(us->out); i++) {
+		if (!short_channel_id_eq(&us->out[i]->short_channel_id,
+					 &schanid))
+			continue;
+
+		update = us->out[i]->channel_update;
```

Comment:
Maybe move this out of the loop down to `reply:`?

Reply:
Yes... I was meaning to get back to this when you merged it anyway.  Oh well..

Reply:
Sorry about that, I thought we can also address these things later and I should not delay this any further :-)